registrations/views.py

```diff
@@ -363,6 +363,72 @@ def get(self, request, *args, **kwargs):
         return Response(resp, status=status)
 
 
+class JembiFacilityCheckHealthcheckView(APIView):
+
+    """ Jembi Facility Check Healthcheck Interaction
+        GET - returns service up - getting auth'd requires DB
+    """
+
+    permission_classes = (IsAuthenticated,)
+
+    def get(self, request, *args, **kwargs):
+        if not (
+            settings.JEMBI_BASE_URL
+            and settings.JEMBI_USERNAME
+            and settings.JEMBI_PASSWORD
+        ):
+
+            return Response(
+                "Jembi integration is not configured properly.",
+                status.HTTP_503_SERVICE_UNAVAILABLE,
+            )
+
+        try:
+            clinic_code = request.query_params.get("clinic_code")
+            result = requests.get(
+                urljoin(settings.JEMBI_BASE_URL, "NCfacilityCheck"),
+                headers={"Content-Type": "application/json"},
+                auth=(settings.JEMBI_USERNAME, settings.JEMBI_PASSWORD),
+                params={"criteria": "value:{}".format(clinic_code)},
+                verify=False,
+                timeout=10.0,
+            )
+            result.raise_for_status()
+
+            jembi_result = result.json()
+
+            if "rows" in jembi_result:
+                resp = {"up": True}
+
+        except (requests.exceptions.HTTPError,) as e:
+            if e.response.status_code == 400:
+                logger.warning(
+                    "400 Error when posting to Jembi.\n"
+                    "Response: %s\nPayload:%s" % (e.response.text)
+                )
+                return Response(
+                    "Error when posting to Jembi. Body: %s Payload: %r"
+                    % (e.response.content),
+                    status=status.HTTP_400_BAD_REQUEST,
+                )
+            else:
+                raise e
+
+        except (requests.exceptions.Timeout,) as e:
+            if e.response.status_code == 504:
```

Comment:
If we have a timeout, then we won't have a response (a timeout means that the server hasn't sent us a response in time)

Reply:
so, I should remove the response from the exception?

Reply:
I think we should just remove the `if`. If we get a `Timeout` exception, then we should just return a 504.

Reply:
I'm concerned that I do not know how to effectively test out and mock a 504 error

Reply:
Yes, that is something that I don't think we can really mock out. We don't want the tests waiting around for 10 seconds for a timeout to happen. I think for this case, it's more work to be able to create a unit test for it than it's worth, so I think we can leave that out for now.